File: Banking-Kata/src/main/java/banking/persistence/NetworkRepository.java
package banking.persistence;

import java.util.HashMap;
import java.util.Optional;
import java.util.Random;
import java.util.UUID;

import banking.model.Identifiable;

public class NetworkRepository<T extends Identifiable> implements Repository<T> {

	private final Random random = new Random();

	private final HashMap<UUID, T> items = new HashMap<UUID, T>();

	@Override
	public Optional<T> findOne(final UUID id) {
		randomlyExplodeSlowly();
		return Optional.of(items.get(id));
	}

	@Override
	public void save(T item) {
		randomlyExplodeSlowly();
		items.put(item.getId(), item);
	}

	private void randomlyExplodeSlowly() {
		try {
			final int sleepTime = random.nextInt(2000) + 3000;
			Thread.sleep(sleepTime);
		} catch (final InterruptedException e) {
			e.printStackTrace();
		}
		final boolean willExplode = random.nextDouble() < 0.1;
		if (willExplode) {
			throw new RuntimeException("The network repository has exploded.");
		}
	}

}

File: 07-Banking-Kata/csharp-dotnetcore2/BankingKataAPI/Persistence/NetworkRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using BankingKataAPI.Models;

namespace BankingKataAPI.Persistence
{
    public class NetworkRepository<T> : IRepository<T>
        where T : class, Identifiable
    {
        private readonly Dictionary<Guid, T> _items;
        private readonly Random _random;

        public NetworkRepository()
        {
            _random = new Random();
            _items = new Dictionary<Guid, T>();
        }

        public T FindOne(Guid id)
        {
            RandomlyExplode();
            return _items.ContainsKey(id)
                ? _items[id]
                : null;
        }

        public void Save(T item)
        {
            RandomlyExplode();
            _items[item.GetId()] = item;
        }

        private void RandomlyExplode()
        {
            
            try {
                var sleepTime = _random.Next(2000) + 1000;
                Thread.Sleep(sleepTime);
            } catch (Exception e)
            {
                Console.WriteLine(e.StackTrace);
            }
            var willExplode = _random.NextDouble() < 0.04;
            if (willExplode) {
                throw new Exception("The network repository has exploded.");
            }
        }
    }
}
